src/controllers/module/fileController.ts

```diff
@@ -32,40 +41,77 @@ export class FileController {
 				res.sendStatus(404)
 			}
 		})
-		this.router.get('/content-management/courses/:courseId/module/module-file', this.getFile())
-		this.router.post('/content-management/courses/:courseId/module/module-file', this.setFile())
+		this.router.get('/content-management/courses/:courseId/module-file', this.getFile())
+		this.router.post('/content-management/courses/:courseId/module-file', this.setFile())
 	}
+
 	public getFile() {
 		return async (request: Request, response: Response) => {
-			response.render('page/course/module/module-file')
+			if (
+				request.session!.sessionFlash &&
+				request.session!.sessionFlash.mediaId &&
+				request.session!.sessionFlash.mediaId != 'undefined'
+			) {
+				const mediaId = request.session!.sessionFlash.mediaId
+
+				const media = await this.restService.get(`/${mediaId}`)
+
+				return response.render('page/course/module/module-file', {media})
+			}
+
+			return response.render('page/course/module/module-file')
 		}
 	}
+
 	public setFile() {
 		return async (request: Request, response: Response) => {
 			const req = request as ContentRequest
 			let data = {
 				...req.body,
 			}
+
+			data.type = fileType.getFileModuleType(data.file)
+
 			const course = response.locals.course
 			let module = await this.moduleFactory.create(data)
-			let errors = await this.moduleValidator.check(data, ['title', 'description', 'file'])
+			let errors = await this.moduleValidator.check(data, ['title', 'description'])
+
+			const mediaId = data.mediaId
+			let file
+			if (mediaId) {
+				file = await this.restService.get(`/${mediaId}`)
```

Comment:
RestService is specific to the learning catalogue rather than a generic http client. It would be better to use a new axios instance here.

Reply:
I thought that the RestService was used as a generic http client, previously something similar was done with the Youtube module, per @elxsy's suggestion